### ISSUE TITLE
Deduplicate setup logic between this and ros-tooling/setup-ros

### ISSUE THREAD

**Issue by emersonknapp** [ros-tooling/setup-ros-docker#31]
The ROS 2 Installation logic is duplicated between this tool and https://github.com/ros-tooling/setup-ros/ - meaning whenever we need to tweak it we have to do it in two places. Deduplicate!

**Comment by flynneva** [comment]
@emersonknapp this script is much faster too compared to `ros-tooling/setup-ros`

**Comment by flynneva** [comment]
the only thing that I see is missing is the support for MacOS/Windows with this repo. I think with the other `ros-tooling/setup_ros` you could use it for Mac/Windows (might be broken, but not due to that repo).

**Comment by emersonknapp** [comment]
Yeah - I don't think that you can run windows containers on Action runners (I might be wrong) - and as far as I know there is no such thing as a "osx container". So, like you said, we would need to continue using the action-based setup for those platforms, rather than jumpstarting from an image like we do with `setup-ros-docker`

**Comment by kenji-miyake** [comment]
As for installing packages, I think parsing `setup-ros` source code and generating `requirements.txt` is a good way.
https://github.com/kenji-miyake/setup-ros-docker/pull/7

(How to change `setup-ros-docker` to use `requirements.txt` is written here.)
https://github.com/ros-tooling/setup-ros-docker/issues/41#issuecomment-906954213

Currently, it only supports `pip3` packages, but it can easily be extended to `apt` packages as well.

With this, I believe it will be easier to make docker images that have corresponding versions to `setup-ros`.
This discussion is held here.
https://github.com/ros-tooling/setup-ros-docker/issues/41#issuecomment-907334301

@emersonknapp @christophebedard How do you feel about it?

**Comment by christophebedard** [comment]
@kenji-miyake that's an interested approach! I know @emersonknapp was working on some tooling to manage common config files across ros-tooling repos: https://github.com/ros-tooling/repo-configs. Not sure if he wanted to use it to manage the `requirements.txt` files.

**Comment by emersonknapp** [comment]
I don't think it makes sense to use `repo-configs` to manage this thing. That mechanism is intended for common configuration of the repositories themselves, and probably shouldn't touch any business logic. It makes more sense to me, to set up a dependency between `setup-ros` and `setup-ros-docker` so that one consumes information from the other.
